### ISSUE TITLE
Need info about running sample project

### ISSUE THREAD

**Issue by vikramvi** [mosclofri/cucumber-appium-java#1]
@mosclofri it will be really helpful if you can please update READ.me with below info

1. does one needs to start appium server before running test cases?
2. what config changes one needs to do on his/her end 
any other info 

Thanks in advance

**Comment by mosclofri** [comment]
Hi @vikramvi ,

I will update READ.me with more detailed info as soon as possible

In quick response to your questions:

- No, you don't need to start Appium server manually. [AppiumServer](https://github.com/mosclofri/cucumber-appium-java/blob/master/testapi/src/main/java/com/appium/api/server/AppiumServer.java) is responsible for starting the server. BUT for proper functioning your appium must be installed with npm. 

 **mvn clean install -D"device.name"="DesiredDeviceName** will execute the entire process. After the tests are done you will find html reports inside android-test/target/html-reports folder. Failing tests will also embed screenshot and device logs to the report file

- You don't need to change anything to make it work. You can edit [android-test](https://github.com/mosclofri/cucumber-appium-java/blob/master/android-test/pom.xml) module's pom.xml as your wish. See between 'systemPropertyVariables' tags. 

 Executing **mvn clean install -D"systemPropertyVariables"="desiredValue"**  you can also override any systemPropertyVariables like these.

**PS:** I have fixed a bug that prevent apk file location detection on Windows env.

**Comment by vikramvi** [comment]
@mosclofri Thanks a ton for detailed info.

**Comment by vikramvi** [comment]
@mosclofri I tried running from Eclipse and face few issues as below

1.  values are being read from class Property

2. Need to hard code them as below

`service = buildService(new AppiumServiceBuilder().
        	                                     withIPAddress("127.0.0.1").
        	                                     usingPort(Integer.parseInt("4723"))
        	       .withAppiumJS(new File("/usr/local/lib/node_modules/appium/build/lib/main.js"))
                       .withArgument(GeneralServerFlag.LOG_LEVEL, "info"));`

3. Need to add 
`withAppiumJS(new File("/usr/local/lib/node_modules/appium/build/lib/main.js"))`

as without this it was throwing an error.

can you please review and fix these as well ?

Thanks,
Vikram

**Comment by mosclofri** [comment]
Hello @vikramvi 

It is running fine for me inside OSX, Windows and Linux with Intellij. I believe its more likely to be problem with your environment settings. Could you able to run the "appium" command directly within bash? I think, code fails to locate Appium inside your PATH and it requires manual file location.

**Comment by vikramvi** [comment]
yes I could run appium command directly from bash and it works well.

I think it may be issue with Eclipse IDE , as I have to explicitly mention ANDROID_HOME  in environment variable in Run Configuration dialog.

Can you run test case from IDE itself instead of command line ?

**Comment by mosclofri** [comment]
Hi @vikramvi 

Does it work through bash via 'mvn clean install' without explicitly mention main.js file? I am not using Eclipse but with Intellij, its works fine when i run RunTest.java over IDE
